### ISSUE TITLE
Please support building against system's iniparser (rather than the bundled one)

### ISSUE THREAD

**Issue by hartwork** [karlstav/cava#61]
Hi!

Some Linux distros do have packages for iniparser, e.g. Gentoo does. It would be cool to be able to build cava against the iniparser installed system-wide rather than the bundled one. For a list  of reasons why that would be helpful, let me point to this page:
https://wiki.gentoo.org/wiki/Why_not_bundle_dependencies

In case a related patch does not seem worth your own time at the moment, please let me know if you would accept a patch in that direction, in general.

Best, Sebastian

**Comment by CelestialWalrus** [comment]
We use a custom iniparser (ndevilla/iniparser). I don't think it's the one available in Gentoo and it works, so replacing it isn't a good idea.

**Comment by karlstav** [comment]
I don't know much about iniparsers, but if there is a system-wide iniparser with packages for all mayor distros (libconfig?), then maybe we should check it out. If not, @CelestialWalrus solution is probably best for now .

**Comment by anko** [comment]
@CelestialWalrus

> We use a custom iniparser (ndevilla/iniparser). I don't think it's the one available in Gentoo

No, that's the same iniparser packaged in Gentoo and Arch. The *upstream URL* link on [Arch's iniparser package](https://www.archlinux.org/packages/extra/x86_64/iniparser/) and the *Homepage* link on [Gentoo's iniparser package](https://packages.gentoo.org/package/dev-libs/iniparser?arches=all) both point to the [project's homepage](http://ndevilla.free.fr/iniparser/), which links to [the github page](https://github.com/ndevilla/iniparser).

> it works, so replacing it isn't a good idea

![think about it](https://cloud.githubusercontent.com/assets/5231746/9097045/67cbc500-3bc1-11e5-873a-87c0d557d827.png)

- - -

For some *actual* counter-points, see the [static linux FAQ](http://sta.li/faq).

I'm in favour, mainly because this allows improvements to happen to the libraries we rely on, without us having to do anything. Iniparser is a really common library, so almost all distros have it packaged ([Ubuntu](https://launchpad.net/~dupondje/+archive/ubuntu/ppa/+build/3940508), [CRUX](https://crux.nu/~lucas/portdb/?a=repo&q=opt), [CentOS](http://rpm.pbone.net/index.php3/stat/4/idpl/23828305/dir/centos_6/com/libiniparser-3.1-1.3.i686.rpm.html), [Fedora](https://admin.fedoraproject.org/pkgdb/package/iniparser/)…). For packagers, having the option to compile against a system library sounds like a good thing.

**Comment by hartwork** [comment]
> We use a custom iniparser (ndevilla/iniparser).

It's the same author, it's not a fork. Looking at http://ndevilla.free.fr/iniparser/index.html you can see
```
git clone http://github.com/ndevilla/iniparser.git
```
up there, pointing to the very same Git repository.

> I don't think it's the one available in Gentoo and it works, so replacing it isn't a good idea.

It soon will be, see https://bugs.gentoo.org/show_bug.cgi?id=556774 and https://github.com/ndevilla/iniparser/issues/62.

**Comment by karlstav** [comment]
valid points.

as I said earlier, if there is a system iniparser available for all major distros we should definitely use it. And if you want to create the patch @hartwork that would be awesome as I am striped from time at the moment. But I can't seem to find iniParser for Ubuntu/debian, am I missing something?

**Comment by hartwork** [comment]
> But I can't seem to find iniParser for Ubuntu/debian, am I missing something?

The link of @anko for ubuntu is an unofficial repository. Officially, Debian and Ubuntu do not have a package for iniparser, afaik. There was a "intent to package" ticket (ITP for short) in Debian but it was closed, see https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=582657#51.

**Comment by karlstav** [comment]
@hartwork ok, how about using [libconfig](http://www.hyperrealm.com/libconfig/)? To me it looks like it does the same as ini parser and I think there is also available packages for all mayor distros. I'll pull your request for now, but I really think we should make things as distro independent as possible.

@CelestialWalrus what do you think? You wrote the original config file stuff for iniparser, do you think it could be "ported" to use libconfig instead? Its [API](http://www.hyperrealm.com/libconfig/libconfig_manual.html#The-C-API) looks pretty straight forward.

**Comment by CelestialWalrus** [comment]
@karlstav I'm busy with projects ATM, so I won't be able to port it now.

**Comment by hartwork** [comment]
> @hartwork ok, how about using libconfig? To me it looks like it does the same as ini parser and I think there is also available packages for all mayor distros. I'll pull your request for now, but I really think we should make things as distro independent as possible.

No objections. If you want to go that road, I would recommend to code against both libconfig 1.4.9 and 1.5 (if their interface differs) so that both users of a stable and bleeding-edge distro (or release of a distro) have what you need. For example on Debian:
```
oldstable  wheezy   1.4.8
stable     jessie   1.4.9
testing    stretch  1.4.9
unstable   sid      1.5
```

**Comment by karlstav** [comment]
this is implemented in 0.4.0
